Modules/_pickle.c

```diff
@@ -1653,7 +1653,7 @@ _Unpickler_SetInputEncoding(UnpicklerObject *self,
 static int
 _Unpickler_SetBuffers(UnpicklerObject *self, PyObject *buffers)
 {
-    if (buffers == NULL) {
+    if (buffers == NULL || buffers == Py_None) {
```

Comment:
Propagating down 2 special values meaning "no buffers" is ugly. Can we 
normalize the value when we parse the arguments, so the rest of the code
have exactly one value for "no buffers"?